frontend/53/autocomplete.js

```diff
@@ -0,0 +1,47 @@
+class AutoComplete {
```

Comment:
The solution is very good overall.
There is no need to keep all the CSS and HTML inside the JS.
Can you move out what you can outside of JS and also give it some styling?